contrib/format-ltsv/src/main/java/org/apache/drill/exec/store/ltsv/LTSVRecordIterator.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.ltsv;
+
+import org.apache.drill.common.exceptions.UserException;
+import org.apache.drill.common.types.TypeProtos;
+import org.apache.drill.exec.physical.resultSet.RowSetLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Iterator;
+
+public class LTSVRecordIterator implements Iterator {
+
+  private static final Logger logger = LoggerFactory.getLogger(LTSVRecordIterator.class);
+
+  private RowSetLoader rowWriter;
+
+  private BufferedReader reader;
+
+  private String line;
+
+  private int recordCount;
+
+  public LTSVRecordIterator(RowSetLoader rowWriter, BufferedReader reader) {
+    this.rowWriter = rowWriter;
+    this.reader = reader;
+
+    // Get the first line
+    try {
+      line = reader.readLine();
+    } catch (IOException e) {
+      throw UserException
+        .dataReadError()
+        .message("Error reading LTSV Data: {}", e.getMessage())
+        .build(logger);
+    }
+  }
+
+  @Override
+  public boolean hasNext() {
+    return line != null;
+  }
+
+  @Override
+  public Boolean next() {
```

Comment:
This does next line? Maybe call it `nextRow` to avoid conflicts with the `next()` which means `nextBatch()`.

Then, simplify (below is abbreviated):
```
try {
  for (;;) {
    String line = reader.readLine();
    if (line == null) { return false; }
    lineNumber++;
    line = line.trim();
    if (line.isEmpty()) { continue; }
    processRow();
  }
} catch (Exception e) {
  UserException...
    .addContext("Line", lineNumber)
}
```